internal/app/app.go

```diff
@@ -41,6 +42,7 @@ type PolicyAutomationApp struct {
 	out       *outputs.Output
 	collector outputs.ValidationResultCollector
 	gke       *gke.GKEClient
+	gkeLocal  *gke.GKELocalClient
```

Comment:
I would introduce an interface for a `GKEClient` and use it for both, local client and actual (remote) client. So the rest of the code logic could just stay as it.
However this solution is valid, the interface can be added in a next step as you wish.

Reply:
I really like the idea of an interface. I'd merge this and open up an issue to track this improvement.